### PR TITLE
#205 Refactor duplicate logic in auth component into one place

### PR DESCRIPTION
**Pull request by PublicFedorenko**
https://trello.com/c/vzl9QhN0/205-refactor-duplicate-logic-in-auth-component-into-one-place

### CODE REVIEW

**Review by olena-kozak — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]

File: frontend/src/app/core/services/auth-form.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+@Injectable({
+    providedIn: 'root',
+})
+export class AuthFormService {
+    constructor(
+        private authService: AuthService,
+        private spinnerService: SpinnerService,
+        private userService: UserService,
+        private notificationService: NotificationService,
+        private router: Router, // eslint-disable-next-line no-empty-function
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth-form.service.ts]: please move this under constructor

File: frontend/src/app/core/services/auth.service.ts [hunk 2]
```diff
@@ -2,117 +2,127 @@ import { Injectable } from '@angular/core';
...
-            );
+    public refreshToken(): Observable<void> {
+        return this.afAuth.authState.pipe(
+            first((u) => !!u),
+            concatMap((u) => this.setUserAccessToken(u!)),
+        );
     }
 
-    public getAccessToken() {
+    public getAccessToken(): string | null {
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts]: why did you add return types? this is not necessary

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/core/services/auth.service.ts (hunk 2 above)
Comment (PublicFedorenko) [reply frontend/src/app/core/services/auth.service.ts]: I'll explain my point. Returning the types explicitly, makes the overall picture clear. When someone needs to check return types of multiple methods, they don't need to hover over each one. Also, it reduces the probability someone will return a type, which is not consistent with other return types in the file (while adding a new feature e.g.)

Should all the explicit return types be removed, or just this concrete one? Removed this one for now

File: frontend/src/app/core/services/auth-form.service.ts (hunk 1 above)
Comment (PublicFedorenko) [reply frontend/src/app/core/services/auth-form.service.ts]: Done

**Review by tatianahutii — APPROVED** [review]
